api/types/types.go

```diff
@@ -34,6 +34,8 @@ type Ping struct {
 	// should use other ways to get the current swarm status, such as the /swarm
 	// endpoint.
 	SwarmStatus *swarm.Status
+
+	EngineFeatures []string
```

Comment:
Opinions on the type here? I'm considering this could match the daemon `features` and be a `map[string]bool`.
@vvoland @thaJeztah @dmcgowan

Reply:
IMO, we should keep the same type.

Not sure if we'd really want to have a feature like this, but be troublesome to convert when a feature is on by default and it needs an explicit opt-out with a `false` value. The conversion would then have to know all possible feature flags and append it to the slice if it's not present in daemon's `features`.

Reply:
Perhaps just "Features" would work to keep it more generic? (potentially this would allow for, e.g., Docker Desktop to append to the list). If we do, we could add some wording / convention to document that the list can be extended, but in that case should use some prefix / namespace (`com.docker.desktop.feature-x=true`, `com.mirantis.feature-y=true`)?

Reply:
I'm okay with that! (I actually had `Features` originally, then changed to `Engine-Features` to make it more explicit, but okay with changing back). That said, we could also keep it `Engine-Features` and still use it for those cases – `Features` seems too generic in the context of HTTP headers and could be confused for something else.

Reply:
@vvoland @dmcgowan thoughts on this?

Reply:
`Features` seems good to me. If we want to be extra safe and avoid clashing with a potential future standard HTTP header, maybe it could be `X-Features` (that wouldn't be consistent with our other headers though).